ngraph/core/reference/src/runtime/reference/reshape.cpp

```diff
@@ -2,46 +2,78 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#include <cmath>
-#include <stdio.h>
+#include <cstring>
+#include <numeric>
 
 #include "ngraph/check.hpp"
+#include "ngraph/coordinate_range.hpp"
+#include "ngraph/coordinate_transform.hpp"
 #include "ngraph/runtime/reference/reshape.hpp"
 
 using namespace ngraph;
 
+namespace
+{
+    template <typename T>
+    T reorder(const T& t, const AxisVector& order)
+    {
+        auto reordered = t;
+        auto out = begin(reordered);
+        NGRAPH_CHECK(t.size() <= order.size());
+        for (size_t i = 0; i < t.size(); ++i)
+        {
+            *out = t[order[i]];
+            ++out;
+        }
+        return reordered;
+    }
+
+    std::vector<int64_t> axis_element_offset(const Shape& s)
+    {
+        std::vector<int64_t> offset(s.size(), 1);
+        for (size_t i = s.size() - 1; i != 0; --i)
+        {
+            offset[i - 1] = offset[i] * s[i];
+        };
+        return offset;
+    }
+
+    void reshape_impl(const char* const arg,
```

Comment:
Why did you add `reshape_impl` function? Looks like you can just rewrite the body of `runtime::reference::reshape` function 😄

Reply:
Looks like encapsulation freak. I'll move impl to `reshape` body.